### DISCUSSION TITLE
Schema generation for list of optional values

### DISCUSSION THREAD

**Discussion by darylweir** [pydantic/pydantic#2576] [Question] [answered]
I have a model like this:
```
class Sample(BaseModel):
    interval: float
    items: List[Optional[float]]
    timestamp: LocalizedDateTime
```

It represents a fixed length sample of a measurement, with one measurement taken every `interval` seconds starting at `timestamp`. Some or all of the measurements may be missing (null), hence the `Optional[float]`.

When I call `Sample.schema()`, I get the following:
```
{
  'title': 'Sample', 
  'type': 'object',
  'properties': 
    {
      'interval': {'title': 'Interval', 'type': 'number'}, 
      'items': {'title': 'Items', 'type': 'array', 'items': {'type': 'number'}},
      'timestamp': {'title': 'Timestamp', 'format': 'LocalizedDateTime', 'type': 'string'}
    }, 
  'required': ['interval', 'items', 'timestamp']
}
```

So the optionality of `items` is not reflected. I would expect `schema['properties']['items']['items']` to be `{ 'type': ['number', 'null'] }`.

Is this a bug or am I thinking about this wrong?

**Comment by PrettyWood** [comment, marked as answer]
Hey @darylweir 
Yep it's a known issue! Here is a possible workaround https://github.com/samuelcolvin/pydantic/issues/1270#issuecomment-729555558. Feel free to have a look at the whole issue!
Hope it helps :)

**Reply by darylweir** [reply]
Thanks! I tried to google for this but apparently my skills failed me.
